File: 5Programs/C#/WSMT-5pb/WSMT-5pb/DataAccess.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace WSMT_5pb
{
    public class DataAccess
    {
        private readonly string filename;

        public DataAccess(string filename)
        {
            this.filename = filename;
        }

        public List<Node> LoadData()
        {
            List<Node> nodes = new List<Node>();
            try
            {
                string[] lines = File.ReadAllLines(filename);

                foreach (string line in lines)
                {
                    string[] splitLine = line.Split(';');
                    Node node = new Node(Int32.Parse(splitLine[0]), Int32.Parse(splitLine[1]), splitLine[2], splitLine[3]);
                    nodes.Add(node);
                }
            }
            catch (IOException e)
            {
                Console.WriteLine("Could not read from file" + e);
            }

            return nodes;
        }
    }
}

File: 5Programs/Python/WSMT-5Prog/DataAccess.py
from typing import List

from Node import Node


class DataAccess:
    def __init__(self, filename):
        self.filename = filename

    def load_data(self):
        nodes: List[Node] = []
        with open(self.filename, "r") as file:
            while True:
                line = file.readline()
                if not line:
                    break
                split_line = line.strip().split(';')
                node = Node(int(split_line[0]), int(split_line[1]), split_line[2], split_line[3])
                nodes.append(node)

        return nodes
